core/src/json_schema.rs

```diff
@@ -19,6 +19,110 @@ pub struct Schema {
     pub any_of: Option<Vec<Schema>>,
 }
 
+impl Schema {
+    // Remove multiple recursively removes Multiple Bson Type entries.
+    pub fn remove_multiple(mut self) -> Result<Self, Error> {
+        // it is invalid for both any_of and bson_type to be set because
+        // any_of is a top level constructor.
+        if !(self.bson_type.is_none() || self.any_of.is_none()) {
```

Comment:
```suggestion
        if self.bson_type.is_some() && self.any_of.is_some() {
```
Personal opinion: positive conditionals are easier to read. "If bson_type and any_of are both Some, then ...", as opposed to "If not (bson_type is None or any_of is None), then ..."